site/app/controllers/forum/ForumController.php

```diff
@@ -879,7 +897,21 @@ private function editPost() {
 
             $markdown = !empty($_POST['markdown_status']);
 
-            return $this->core->getQueries()->editPost($original_creator, $current_user, $post_id, $new_post_content, $anon, $markdown);
+            $current_attachments = explode("\n", $original_post['attachment_name']);
+            if (isset($_POST['deleted_attachments'])) {
+                foreach (json_decode($_POST['deleted_attachments']) as $img) {
+                    if (($key = array_search($img, $current_attachments)) !== false) {
```

Comment:
Assigning variables in an if-statement is error-prone, and can be tricky to read.  It might be better to create a temporary variable above the if-statement.  Note: this is just a suggestion, not necessarily a requested change.